ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -224,8 +232,26 @@ public NettyServerStreamRpc(RaftServer server) {
         .childOption(ChannelOption.SO_KEEPALIVE, true)
         .bind(port);
     this.proxies = new Proxies(new PeerProxyMap<>(name, peer -> newClient(peer, properties)));
-    this.executorService = Executors.newFixedThreadPool(
-        RaftServerConfigKeys.DataStream.asyncThreadPoolSize(server.getProperties()));
+
+    int threadPoolSize = RaftServerConfigKeys.DataStream.asyncThreadPoolSize(server.getProperties());
+    for (int i = 0; i < threadPoolSize; i ++) {
+      this.streamExecutors.add(Executors.newFixedThreadPool(1));
+    }
+  }
+
+  private synchronized ExecutorService getMinTaskExecutor() {
+    ExecutorService minTaskExecutor = null;
+    int minTaskNum = Integer.MAX_VALUE;
+
+    for (ExecutorService e : streamExecutors) {
+      ThreadPoolExecutor tpe = (ThreadPoolExecutor) e;
+      if (minTaskNum > tpe.getQueue().size()) {
+        minTaskNum = tpe.getQueue().size();
+        minTaskExecutor = e;
+      }
+    }
+
+    return minTaskExecutor;
   }
```

Comment:
We should just use the executor service but not try to do thread scheduling ourselves.  The existing ExecutorService is good enough.